disk-utils/cfdisk.c

```diff
@@ -2735,6 +2742,8 @@ int main(int argc, char *argv[])
 	cf->cxt = fdisk_new_context();
 	if (!cf->cxt)
 		err(EXIT_FAILURE, _("failed to allocate libfdisk context"));
+        if (read_only)
+                cf->cxt->readonly = 1;
 
```

Comment:
Don't access `cxt` struct members. The only way how to set it is the 3rd argument of fdisk_assign_device().